### PR TITLE
GT-1448 Adding TranslationsRepository and deprecating TranslationsDownloader

### PR DESCRIPTION
**Pull request by levieggertcru**
This PR starts some of the work for moving to the new translation file downloading api for downloading files individually.  This will be fully completed in one additional PR.  This PR does the following:
- Adds the TranslationsRepository class which is responsible for fetching translation files and translation objects.  This is also utilizing Combine.
- Deprecates TranslationDownloader which contains the old logic for downloading translation zip file data and storing those to disk.  This logic is now handled in TranslationsRepository using Combine.
- Deprecates SHA256FilesCache.  This class was originally created as a common file cacheing class and is not needed because it does a lot of the same things the common FileCache class.
- Deprecates TranslationsFileCache which is a wrapper around ResourcesSHA256FileCache.  Moving this logic directly into the ResourcesSHA256FileCache class for cacheing translation files.  
- Updated some places to point to ResourcesSHA256FileCache instead of TranslationsFileCache.
- Deprecates AttachmentsFileCache which is a wrapper around ResourcesSHA256FileCache.  Will be moving logic for cacheing attachment files into the ResourcesSHA256FileCache class.
- Various changes to the FileCache class.  Added an extension for the SSZipArchive dependency and also added an extension that wraps some of the concrete methods with Combine publishers.
- Changes to MobileContentTranslationsApi to utilize (Combine) URLSession Publisher for making http requests and deprecated some methods in there which can be removed once TranslationDownloader class is removed.

### CODE REVIEW

File: godtools/App/Services/Articles/Repository/Cache/ArticleAemCache.swift [hunk 1]
```diff
@@ -207,7 +207,7 @@ class ArticleAemCache {
     
     private func storeWebArchivePlistData(webArchiveFilename: String, webArchivePlistData: Data) -> ArticleAemCacheError? {
         
-        let cacheFileResult: Result<URL, Error> = fileCache.cache(
+        let cacheFileResult: Result<URL, Error> = fileCache.storeFile(
             location: ArticleAemWebArchiveFileCacheLocation(filename: webArchiveFilename),
```
[the remark this answers is not in the document]
Comment (wrandall22) [reply godtools/App/Services/Articles/Repository/Cache/ArticleAemCache.swift:211]: AEM = Adobe Experience Manager (a content management system)
JCR = a visual representation of the file system for AEM

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Services/Articles/Repository/Cache/ArticleAemCache.swift:211]: Thanks @wrandall22!  I remember the implementation for this was pretty complex and had a lot of pieces.  I'd love to refactor this in the future to use Combine which really does well at streamlining async tasks.

**Review by rachaelblue — COMMENTED** [review]
There's a lot here 😅 Would probably be a good idea to write some unit tests for this at some point.

File: godtools/App/Share/Common/FileCache/FileCache+Combine.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import Combine
+
+extension FileCache {
+    
+    func getFileExistsPublisher(location: FileCacheLocation) -> AnyPublisher<Bool, Error> {
+        
+        switch getFileExists(location: location) {
+        case .success(let fileExists):
+            return Just(fileExists).setFailureType(to: Error.self)
+                .eraseToAnyPublisher()
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Share/Common/FileCache/FileCache+Combine.swift:19]: Hey @rachaelblue I just realized something, Swift Result has a publisher instance on it (https://developer.apple.com/documentation/swift/result/publisher-swift.property).  I think most Swift Standard classes will provide a publisher for us already. I'm going to update this.

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Share/Common/FileCache/FileCache+Combine.swift:19]: Definitely need a lot of unit tests in places.  That is one area that is lacking at the moment.

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Share/Common/FileCache/FileCache+Combine.swift]: Just updated this and removed the FileCache+Combine extension.

Anywhere we're using Combine then we will have to use a Publisher.  I think most Swift Standard classes will provide publishers already which will help having to create code like we saw in the FileCache+Combine extension :D

File: godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        baseUrl = config.mobileContentApiBaseUrl
+    }
+    
+    // MARK: - Files
+    
+    private func getTranslationFileRequest(fileName: String) -> URLRequest {
+        
+        return requestBuilder.build(
+            session: session,
+            urlString: baseUrl + "/translations/files/" + fileName,
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift:31]: Hey @rachaelblue are you saying a separate constants file that holds all of the endpoints? Or in this class?

[the remark this answers is not in the document]
Comment (rachaelblue) [reply godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift:31]: In this class, so it's clear what the paths are supposed to look like.  Just an idea, definitely more complicated than you currently have it, but harder to mess up in the future.
```
enum Paths {
   static let translations = "/translations"
   static let translationsZipFile = translations + "/%@"
   static let translationFile = translations + "/files/%@"
}
```

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift:31]: I see what you're saying, I'm going to make a note of this one and come back to it as I work on moving some more of the apis to Combine.

**Review by rachaelblue — APPROVED** [review]

**Pull request merged**